display_applications/qiime/qiime2/q2view.xml

```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0"?>
+<display id="q2view" version="1.0.0" name="view at">
+    <dynamic_links from_data_table="q2view_display" skip_startswith="#" id="value" name="name">
+        <url>${ url % { 'q2view_file_url_qp': $q2view_file.qp } }</url>

```

Comment:
How does this work? Is the `url % {` mapping something (like URI-encoding) over the object? Or are we assigning into a scope of some sort?

Reply:
`url` comes from the 'url' column of the `q2view_display.loc.sample` file. This is just doing standard dictionary-based string replacement.

The `.qp` attribute of `$q2view_file` calls `urllib.parse.quote_plus` on the URL generated for q2view_file.

Reply:
> This is just doing standard dictionary-based string replacement.

oh. duh! Thanks!

> The .qp attribute of $q2view_file calls urllib.parse.quote_plus on the URL generated for q2view_file.

Ah, neat!